src/datatypes/datetime.js

```diff
@@ -342,6 +342,25 @@ class DateTime {
     return result;
   }
 
+  getPrecisionValue() {
+    const precisionValueMap = new Map();
+    if (this.isTime()) {
+      precisionValueMap.set(DateTime.Unit.HOUR, 2);
+      precisionValueMap.set(DateTime.Unit.MINUTE, 4);
+      precisionValueMap.set(DateTime.Unit.SECOND, 6);
+      precisionValueMap.set(DateTime.Unit.MILLISECOND, 9);
+    } else {
+      precisionValueMap.set(DateTime.Unit.YEAR, 4);
+      precisionValueMap.set(DateTime.Unit.MONTH, 6);
+      precisionValueMap.set(DateTime.Unit.DAY, 8);
+      precisionValueMap.set(DateTime.Unit.HOUR, 10);
+      precisionValueMap.set(DateTime.Unit.MINUTE, 12);
+      precisionValueMap.set(DateTime.Unit.SECOND, 14);
+      precisionValueMap.set(DateTime.Unit.MILLISECOND, 17);
+    }
+    return precisionValueMap.get(this.getPrecision());
+  }
```

Comment:
I think it would be more performant to initialize these maps as constants in the file -- and then just do something like:
```js
const pvMap = this.isTime() ? TIME_PRECISION_VALUE_MAP : DATETIME_PRECISION_VALUE_MAP;
return pvMap.get(this.getPrecision());
```

Reply:
I actually did something like this initially, and then realized that for some reason the `DateTime` namespace was out of scope of where I was defining them at the top of the file. Maybe I can define the maps within the classes themselves, and that should take care of the issue. I'll give it a shot!

Reply:
Turns out the issue was an ordering thing; I put them at the top of the file before `DateTime.Unit` was available. I just put them with the other similar constants and now it's all working.